app/controllers/Application.java

```diff
@@ -1477,12 +1517,41 @@ public static Result restFlowMetricsGraphData(String flowDefId) {
         .order()
         .desc(AppResult.TABLE.FINISH_TIME)
         .setMaxRows(JOB_HISTORY_LIMIT)
-        // The 2nd and 3rd table are not required for plotting the graph
-        //.fetch(AppResult.TABLE.APP_HEURISTIC_RESULTS, AppHeuristicResult.getSearchFields())
-        //.fetch(AppResult.TABLE.APP_HEURISTIC_RESULTS + "."
-        //    + AppHeuristicResult.TABLE.APP_HEURISTIC_RESULT_DETAILS, "*")
         .findList();
 
     return results;
   }
+
+  private static class AppResourceUsageData {
+    String user;
+    long resourceUsed;
+    long resourceWasted;
+  }
+
+  /**
+   * Returns the list of users with their resourceUsed and resourceWasted Data for the given time range
+   * @return list of AppResourceUsageData
+   **/
+  private static Collection<AppResourceUsageData> getUserResourceUsage(Date start, Date end) {
+    long resourceUsed = 0;
+    Map<String, AppResourceUsageData> userResourceUsage = new HashMap<String, AppResourceUsageData>();
+    // Fetch all the appresults for the given time range.
+    List<AppResult> results = AppResult.find.select("*")
+        .where()
+        .ge(AppResult.TABLE.START_TIME, start.getTime())
+        .le(AppResult.TABLE.START_TIME, end.getTime()).findList();
```

Comment:
When we specify start and end date(yyyy-MM-dd), are we excluding jobs which ran on end-date?

Currently, it looks like we are excluding the jobs as getTime for say, 2016-08-03 will return 2016-08-03-00.

Usually when we specify ranges we include from and exclude to, like [from, to) but I am not sure how should the behavior be in our case.